modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -1,20 +1,208 @@
-import { BaseTransaction } from '../baseCoin';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BaseTransaction, TransactionType } from '../baseCoin';
 import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { InvalidTransactionError, SigningError } from '../baseCoin/errors';
+import { construct, decode } from '@substrate/txwrapper-polkadot';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { decodeAddress } from '@polkadot/keyring';
+import { KeyPair } from './keyPair';
+import {
+  TxData,
+  DecodedTx,
+  TransferArgs,
+  StakeArgs,
+  StakeArgsPayeeRaw,
+  AddProxyArgs,
+  ProxyArgs,
+  UnstakeArgs,
+} from './iface';
+import utils from './utils';
 
 export class Transaction extends BaseTransaction {
+  protected _dotTransaction: UnsignedTransaction;
+  private _signedTransaction?: string;
+  private _registry: TypeRegistry;
+  private _chainName: string;
+  private _sender: string;
+
+  constructor(coinConfig: Readonly<CoinConfig>) {
+    super(coinConfig);
+  }
+
   /** @inheritdoc */
-  canSign(key: BaseKey): boolean {
-    throw new NotImplementedError('canSign not implemented');
+  canSign({ key }: BaseKey): boolean {
+    const kp = new KeyPair({ prv: key });
+    const addr = kp.getAddress();
+    if (addr === this._sender) {
```

Comment:
Which is sender initial value?
Could be infer sender when it is signed?

Reply:
No, sender has to be specified when creating the builder